pkg/config/secret-sync-config_test.go

```diff
@@ -14,18 +14,213 @@ limitations under the License.
+			os.Exit(1)
+		}
+		secretManagerClient, err := client.NewSecretManagerClient(context.Background())
+		if err != nil {
+			fmt.Printf("Fail to create new Secret Manager client: %s", err)
+			os.Exit(1)
+		}
+
+		testClient = &tests.E2eTestClient{
+			&client.Client{
+				K8sClientset:        *k8sClientset,
+				SecretManagerClient: *secretManagerClient,
+			},
+		}
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestWatchConfig(t *testing.T) {
+	if !testOpts.e2e {
+		return
+	}
+
+	var testcases = []struct {
+		name          string
+		oldConfigFile string
+		oldConfig     SecretSyncConfig
+		newConfigFile string
+		newConfig     SecretSyncConfig
+	}{
+		{
+			name: "Update ConfigMap.",
+			oldConfigFile: `
+              specs:
+              - source: 
+                  project: proj-1
+                  secret: secret-1
+                destination:
+                  namespace: ns-1
+                  secret: secret-1
+                  key: key-1
+              - source: 
+                  project: proj-1
+                  secret: secret-2
+                destination:
+                  namespace: ns-2
+                  secret: secret-2
+                  key: key-2
+              `,
+			oldConfig: SecretSyncConfig{
+				Specs: []SecretSyncSpec{
+					{
+						Source: SecretManagerSpec{
+							Project: "proj-1",
+							Secret:  "secret-1",
+						},
+						Destination: KubernetesSpec{
+							Namespace: "ns-1",
+							Secret:    "secret-1",
+							Key:       "key-1",
+						},
+					},
+					{
+						Source: SecretManagerSpec{
+							Project: "proj-1",
+							Secret:  "secret-2",
+						},
+						Destination: KubernetesSpec{
+							Namespace: "ns-2",
+							Secret:    "secret-2",
+							Key:       "key-2",
+						},
+					},
+				},
+			},
+			newConfigFile: `
+              specs:
+              - source: 
+                  project: proj-0
+                  secret: secret-0
+                destination:
+                  namespace: ns-0
+                  secret: secret-0
+                  key: key-0
+              `,
+			newConfig: SecretSyncConfig{
+				Specs: []SecretSyncSpec{
+					{
+						Source: SecretManagerSpec{
+							Project: "proj-0",
+							Secret:  "secret-0",
+						},
+						Destination: KubernetesSpec{
+							Namespace: "ns-0",
+							Secret:    "secret-0",
+							Key:       "key-0",
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testcases {
+		testname := tc.name
+		t.Run(testname, func(t *testing.T) {
+
+			configAgent := &Agent{}
+			runFunc, err := configAgent.WatchConfig(testOpts.configPath)
+			if err != nil {
+				t.Error(err)
+			}
+
+			if runFunc != nil {
+				ctx, cancel := context.WithCancel(context.Background())
+				go runFunc(ctx)
+				defer cancel()
+			}
+
+			// initialize the configMap, and wait for 30 seconds for the mounted configMap to be updated.
+			err = testClient.UpsertKubernetesConfigMap("default", "config", "syncConfig", tc.oldConfigFile)
```

Comment:
These are hardcodes, they should be coming in from flags if you want to take this all the way.  What happens if I ran this in a pod in a different namespace, and the configmap mounted in was named `monkeys`?